src/OrleansPSUtils/StartGrainClient.cs

```diff
@@ -0,0 +1,92 @@
+        public ClientConfiguration Config { get; set; }
+
+        [Parameter(Position = 4, Mandatory = true, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public IPEndPoint GatewayAddress { get; set; }
+
+        [Parameter(Position = 5, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public bool OverrideConfig { get; set; } = true;
+
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = FilePathSet)]
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = FileSet)]
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = ConfigSet)]
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public TimeSpan Timeout { get; set; } = TimeSpan.Zero;
+
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                WriteVerbose($"[{DateTime.UtcNow}] Initializing Orleans Grain Client");
+
+                switch (ParameterSetName)
+                {
+                    case FilePathSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using config file at '{ConfigFilePath}'...");
+                        if (string.IsNullOrWhiteSpace(ConfigFilePath))
+                            throw new ArgumentNullException(nameof(ConfigFilePath));
+                        GrainClient.Initialize(ConfigFilePath);
+                        break;
+                    case FileSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using provided config file...");
+                        if (ConfigFile == null)
+                            throw new ArgumentNullException(nameof(ConfigFile));
+                        GrainClient.Initialize(ConfigFile);
+                        break;
+                    case ConfigSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using provided 'ClientConfiguration' object...");
+                        if (Config == null)
+                            throw new ArgumentNullException(nameof(Config));
+                        GrainClient.Initialize(Config);
+                        break;
+                    case EndpointSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using default Orleans Grain Client initializer");
+                        if (GatewayAddress == null)
+                            throw new ArgumentNullException(nameof(GatewayAddress));
+                        GrainClient.Initialize(GatewayAddress, OverrideConfig);
+                        break;
+                    default:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using default Orleans Grain Client initializer");
+                        GrainClient.Initialize();
+                        break;
+                }
+
+                if (Timeout != TimeSpan.Zero)
+                    GrainClient.SetResponseTimeout(Timeout);
+            }
```

Comment:
It is the same principle of `DbConnection`. You can call `Close()`, you can use `using()` and you can call `Dispose()`. All of these are ways for you to close a connection. You still need do it.

Can we do a timeout? Yes, we can by using PSSession. But that would require a lot more of complexity. Also, when the powershell Session is over, the objects are cleaned up. This is a behaviour of the powershell runtime and its session.